File: C-Programming/Practice/Basic C/prime.c
#include <stdio.h>

int i, temp = 0;

void check_prime();
void table();

int main(int argc, char const *argv[])
{
    int num;
    printf("\n Enter Postive Number to find Number is Prime : ");
    scanf("%d",&num);
    check_prime(num);
    printf("\n");
    return 0;
}

void check_prime(number)
{
    if (number > 1) {
        for ( i = 2; i < number; i++)
        {
            if (number % i == 0)
            {
                temp = 1;
                break;
            }
            else {
                temp = 0;
            }
        }
        if (temp == 0)
        {
            printf("\n %d is a Prime Number\n\n\n",number);
            table(number);
        }
        else {
            printf("\n %d is NOT a Prime Number",number);
        }
    }
    else {
        printf("\n Error");
    }
}

void table(num)
{
    for ( i = 1; i < 11; i++)
    {
        printf("\n  %d  *  %d  =  %d",num,i,num*i);
    }
}

File: JAVA/Practice/Basics/prime.java
import java.util.*;

public class prime {
    void table(int n){
        for (int i = 1; i < 11; i++) {
            System.out.println("\t"+n+" * "+i+" = "+(n*i));
        }
    }

    void check_prime(int num){
        int temp = 0;
        if (num > 1){
            for (int i = 2; i < num; i++) {
                if (num % i == 0) {
                    temp = 1;
                    break;
                }
                else {
                    temp = 0;
                }
            }
            if (temp == 0) {
                System.out.println("\n "+num+" is a Prime Number\n\n\n");
                table(num);
            }
            else {
                System.out.println("\n "+num+" is a NOT Prime Number");
            }
        }
    }

    public static void main(String[] args) {
        Scanner ar = new Scanner(System.in);
        System.out.print("\n Enter Positive Number to check Prime Number : ");
        int number = ar.nextInt();
        ar.close();
        prime obj = new prime();
        obj.check_prime(number);
    }
}
